src/main/java/org/broadinstitute/consent/http/models/PaginationToken.java

```diff
@@ -91,16 +86,29 @@ public String getSortDirection() {
     return this.sortDirection;
   }
 
-  public List<String> getFilterTerms() {
-    return this.filterTerms;
+  public String getFilterTerm() {
+    return this.filterTerm;
   }
 
   public Integer getFilteredCount() {
     return this.filteredCount;
   }
-  
+
   public void setFilteredCount(Integer filteredCount) {
     this.filteredCount = filteredCount;
+    /*
+     * Three cases in determining filtered page count
+     * 1. The count is less than the page size: page size = 1
+     * 2. The count/page size modulo page size is 0, i.e. count of 50, page size of 10, page count is 5
+     * 3. Anything else, i.e. count of 55, page size of 10, page size is 6
+     */
+    if (filteredCount <= this.getPageSize()) {
```

Comment:
For pageSize (or any user submitted Integer for that matter), do we want to catch and throw a BadRequestException for values that are 0 or less?

Reply:
We can do something like that. This class should be able to validate itself and if it's not valid, then the resource class should do exactly that. That would prevent bad data from making its way to the service class. I suggest we handle that when we write the API that handles parsing a `PaginationToken`.